src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -634,11 +634,13 @@ public void run()
          if (idleTimeout > 0L) {
             logPoolState("Before cleanup\t");
             final List<PoolBagEntry> bag = connectionBag.values(STATE_NOT_IN_USE);
-            int removable = bag.size() - config.getMinimumIdle();
-            for (PoolBagEntry bagEntry : bag) {
+            final int bagSize = bag.size();
```

Comment:
'for-each' is nice to read. and its faster 'only' for list with no random access.